### ISSUE TITLE
Synchronize supported Python versions with Python support schedule

### ISSUE THREAD

**Issue by stefan6419846** [bo4e/BO4E-python#719]
Since https://github.com/bo4e/BO4E-python/commit/56123cf01fdcb2fac671811ee98ee2a44fdda902, only Python 3.10 and Python 3.11 are supported officially, with Python 3.12 being mentioned in #687 already.

It would be nice to synchronize the supported Python versions with the Python support schedule. At the moment, this would mean that Python 3.8 through Python 3.12 are supported. This might make it easier to use the package in different setups, for example where Python 3.10 is not yet readily available to due other circumstances.

**Comment by hf-krechan** [comment]
Yes that's a good idea. I will create a PR which expands the python version matrix in the GitHub actions.
So all tests will run for each Python version.

**Comment by hf-krechan** [comment]
I created a PR #720 but it seems that we need to make some changes to get it running for Python 3.8 and 3.9.
Do you need these lower versions?

**Comment by stefan6419846** [comment]
You might need to add `from __future__ import annotations` to `test_zusatz_attribut.py` and `serialization_helper.py` - this should fix the tests for Python 3.9 (at least on my local setup).

`typing.Annotated` is new in Python 3.9 (https://docs.python.org/3.9/library/typing.html#typing.Annotated) and thus most likely requires the backports for Python 3.8.

For me, it would be sufficient to add support for Python 3.9 (this is what I currently use and would consider for further evaluation), although I tend to prefer packages which support the full range of officially supported Python versions as in my initial request.

**Comment by hf-krechan** [comment]
I will give it another try next week.
Thanks for the idea with the `annotations` import :)

**Comment by hf-kklein** [comment]
Hey @stefan6419846:
> You might need to add from __future__ import annotations to test_zusatz_attribut.py and serialization_helper.py - this should fix the tests for Python 3.9 (at least on my local setup).

Version [v202401.0.3](https://github.com/bo4e/BO4E-python/releases/tag/v202401.0.3) does exactly that and supports Python 3.9.

Thanks for your contribution!

As you already noted, more effort is required to support Python 3.8. Feel free to open a PR for that, though. EOL is in 6 months, so hopefully no one starts building new applications using 3.8 :)

**Comment by stefan6419846** [comment]
As mentioned previously, Python 3.9 is currently sufficient for my use case, thus I am going to  close this issue.

**Comment by hf-kklein** [comment]
Great, we're happy to hear how you're using bo4e :)
